corehq/apps/locations/adjacencylist.py

```diff
@@ -161,69 +160,52 @@ def make_cte_query(cte):
 
     def _cte_get_queryset_descendants(self, *args, **kw):
         return self._cte_get_descendants(*args, **kw)
 
-    def _mptt_get_queryset_ancestors(self, node, *args, **kw):
-        if isinstance(node, Q):
-            node = self.filter(node)
-        return super(AdjListManager, self).get_queryset_ancestors(node, *args, **kw)
-
-    def _mptt_get_queryset_descendants(self, node, *args, **kw):
-        if isinstance(node, Q):
-            node = self.filter(node)
-        return super(AdjListManager, self).get_queryset_descendants(node, *args, **kw)
+    def root_nodes(self):
+        parent_col = self.model.parent_id_attr
+        return self.all().filter(**{parent_col + "__isnull": True})
 
     def get_queryset_ancestors(self, queryset, include_self=False):
         timing = TimingContext("get_queryset_ancestors")
-        mptt_qs = cte_qs = queryset
+        cte_qs = queryset
         if isinstance(queryset, ComparedQuerySet):
-            mptt_qs = queryset._mptt_set
             cte_qs = queryset._cte_set
-        if settings.IS_LOCATION_CTE_ONLY:
-            mptt_set = None
-        else:
-            with timing("mptt"):
-                mptt_set = self._mptt_get_queryset_ancestors(mptt_qs, include_self)
-        if settings.IS_LOCATION_CTE_ENABLED:
-            with timing("cte"):
-                cte_set = self._cte_get_queryset_ancestors(cte_qs, include_self)
-        else:
-            cte_set = None
+        mptt_set = None
+        with timing("cte"):
+            cte_set = self._cte_get_queryset_ancestors(cte_qs, include_self)
         return ComparedQuerySet(mptt_set, cte_set, timing)
 
     def get_queryset_descendants(self, queryset, include_self=False):
         timing = TimingContext("get_queryset_descendants")
-        mptt_qs = cte_qs = queryset
+        cte_qs = queryset
         if isinstance(queryset, ComparedQuerySet):
-            mptt_qs = queryset._mptt_set
             cte_qs = queryset._cte_set
-        if settings.IS_LOCATION_CTE_ONLY:
-            mptt_set = None
-        else:
-            with timing("mptt"):
-                mptt_set = self._mptt_get_queryset_descendants(mptt_qs, include_self)
-        if settings.IS_LOCATION_CTE_ENABLED:
-            with timing("cte"):
-                cte_set = self._cte_get_queryset_descendants(cte_qs, include_self)
-        else:
-            cte_set = None
+        mptt_set = None
+        with timing("cte"):
+            cte_set = self._cte_get_queryset_descendants(cte_qs, include_self)
         return ComparedQuerySet(mptt_set, cte_set, timing)
 
 
-class AdjListModel(MPTTModel):
+class FakeMPTTModel(models.Model):
+    """Add defaults for MPTT fields until they can be dropped"""
+
+    class Meta:
+        abstract = True
+
+    tree_id = models.IntegerField(default=0)
+    lft = models.IntegerField(default=0)
+    rght = models.IntegerField(default=0)
+    level = models.IntegerField(default=0)
```

Comment:
I'm planning to drop these fields and referencing indexes sometime later once we are confident that we will not need them. At this point delayed removal is just extra precaution, nothing more.